server/lib/src/idm/server.rs

```diff
@@ -1379,6 +1379,10 @@ impl<'a> IdmServerAuthTransaction<'a> {
                 effective_session: LdapSession::UnixBind(UUID_ANONYMOUS),
             }))
         } else {
+            if self.qs_read.d_info.d_ldap_allow_unix_pw_bind == false {
```

Comment:
clippy will complain, this should be `if !self.qs_read.d_info.d_ldap_allow_unix_pw_bind { `